wlauto/workloads/multiapp/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java

```diff
@@ -0,0 +1,141 @@
+package com.arm.wlauto.uiauto.multiapp;
+
+import android.os.Bundle;
+
+// Import the uiautomator libraries
+import com.android.uiautomator.core.UiObject;
+import com.android.uiautomator.core.UiScrollable;
+import com.android.uiautomator.core.UiSelector;
+
+import com.arm.wlauto.uiauto.UxPerfUiAutomation;
+
+import static com.arm.wlauto.uiauto.BaseUiAutomation.FindByCriteria.BY_ID;
+import static com.arm.wlauto.uiauto.BaseUiAutomation.FindByCriteria.BY_TEXT;
+import static com.arm.wlauto.uiauto.BaseUiAutomation.FindByCriteria.BY_DESC;
+
+import java.util.concurrent.TimeUnit;
+
+public class UiAutomation extends UxPerfUiAutomation {
+
+    // Create UIAutomation objects
+    private com.arm.wlauto.uiauto.googlephotos.UiAutomation googlephotos =
+        new com.arm.wlauto.uiauto.googlephotos.UiAutomation();
+
+    private com.arm.wlauto.uiauto.gmail.UiAutomation gmail =
+        new com.arm.wlauto.uiauto.gmail.UiAutomation();
+
+    private com.arm.wlauto.uiauto.skype.UiAutomation skype =
+        new com.arm.wlauto.uiauto.skype.UiAutomation();
+
+    public Bundle parameters;
+    public String packageName;
+    public String packageID;
+
+    public void runUiAutomation() throws Exception {
+        parameters = getParams();
+        packageName = parameters.getString("package");
+        packageID = packageName + ":id/";
+
+        // Setup the three uiautomator classes with the correct information
+        // Also create a dummy parameter to disable marker api as they
+        // should not log actions themselves.
+        Bundle dummyParams = new Bundle();
+        dummyParams.putString("markers_enabled", "false");
+        googlephotos.parameters = dummyParams;
+        googlephotos.packageName = parameters.getString("googlephotos_package");
+        googlephotos.packageID = googlephotos.packageName + ":id/";
+        gmail.parameters = dummyParams;
+        gmail.packageName = parameters.getString("gmail_package");
+        gmail.packageID = gmail.packageName + ":id/";
+        skype.parameters = dummyParams;
+        skype.packageName = parameters.getString("skype_package");
+        skype.packageID = skype.packageName + ":id/";
+
+        String recipient = parameters.getString("recipient");
+        String loginName = parameters.getString("my_id");
+        String loginPass = parameters.getString("my_pwd");
+        String contactName = parameters.getString("name").replace("0space0", " ");
+
+        setupGooglePhotos();
+        sendToGmail(recipient);
+        logIntoSkype(loginName, loginPass);
+        // Skype won't allow us to login and share on first visit so invoke
+        // once more from googlephotos
+        pressBack();
+        pressBack();
+        sendToSkype(contactName);
+
+        unsetScreenOrientation();
+    }
+
+    private void setupGooglePhotos() throws Exception {
+        sleep(5); // Pause while splash screen loads
```

Comment:
Is a fixed delay necessary here? Can you not wait for the splash screen UI  element to first appear and then disappear here?